include/ignition/gui/qml/GzPose.qml

```diff
@@ -0,0 +1,291 @@
+ *  users can read from signal pararmeters of gzPoseSet: _x, _y, etc.
+ *
+ *  Usage example:
+ *  GzPose {
+ *    id: gzPose
+ *    readOnly: false
+ *    xValue: xValueFromCPP
+ *    yValue: yValueFromCPP
+ *    zValue: zValueFromCPP
+ *    rollValue: rollValueFromCPP
+ *    pitchValue: pitchValueFromCPP
+ *    yawValue: yawValueFromCPP
+ *    onGzPoseSet: {
+ *      myFunc(_x, _y, _z, _roll, _pitch, _yaw)
+ *    }
+ *  }
+**/
+
+Item {
+  id: gzPoseRoot
+
+  // Read-only / write
+  property bool readOnly: false
+
+  // User input value.
+  property double xValue
+  property double yValue
+  property double zValue
+  property double rollValue
+  property double pitchValue
+  property double yawValue
+
+  /**
+   * Used to read spinbox values
+   * @params: _x, _y, _z, _roll, _pitch, _yaw: corresponding spinBoxes values
+   * @note: When readOnly == false, user should read spinbox value from its
+   *        parameters.
+   *        When readOnly == true, this signal is unused.
+   */
+  signal gzPoseSet(double _x, double _y, double _z, double _roll, double _pitch, double _yaw)
+
+
+  /*** The following are private variables: ***/
+  // Show Pose bar (used to control expand)
+  property bool show: true
+
+  height: gzPoseContent.height
+
+  // Left indentation
+  property int indentation: 10
+
+  // Horizontal margins
+  property int margin: 5
+
+  // Maximum spinbox value
+  property double spinMax: 1000000
+
+  // local variables to store spinbox values
+  property var xItem: {}
+  property var yItem: {}
+  property var zItem: {}
+  property var rollItem: {}
+  property var pitchItem: {}
+  property var yawItem: {}
+
+  // Dummy component to use its functions.
+  IgnHelpers {
+    id: gzHelper
+  }
+  /*** Private variables end: ***/
+
+  /**
+   * Used to create a spin box
+   */
+  Component {
+    id: writableNumber
+    IgnSpinBox {
+      id: writableSpin
+      value:  numberValue
+      minimumValue: -spinMax
+      maximumValue: spinMax
+      decimals: gzHelper.getDecimals(writableSpin.width)
+      onEditingFinished: {
+        gzPoseRoot.gzPoseSet(xItem.value, yItem.value, zItem.value, rollItem.value, pitchItem.value, yawItem.value)
+      }
+    }
+  }
+
+  /**
+   * Used to create a read-only number
+   */
+  Component {
+    id: readOnlyNumber
+    Text {
+      id: numberText
+      anchors.fill: parent
+      horizontalAlignment: Text.AlignRight
+      verticalAlignment: Text.AlignVCenter
+      text: {
+        var decimals = gzHelper.getDecimals(numberText.width)
+        return numberValue.toFixed(decimals)
+      }
+    }
+  }
+
+  Rectangle {
+    id: gzPoseContent
+    width: parent.width
+    height: show ? gzPoseGrid.height : 0
+    clip: true
+    color: "transparent"
+
+    Behavior on height {
+      NumberAnimation {
+        duration: 200;
+        easing.type: Easing.InOutQuad
+      }
+    }
+
+    GridLayout {
+      id: gzPoseGrid
+      width: parent.width
+      columns: 6
+
+      // Left spacer
```

Comment:
Related to the indentation and margin comment above, I think we can leave these spacers out of the reusable widget and instead let the parent pad this widget as they see fit.